zlib-rs/src/allocate.rs

```diff
@@ -251,7 +270,11 @@ impl Allocator<'_> {
     fn allocate_layout_zeroed(&self, layout: Layout) -> *mut c_void {
         #[cfg(feature = "rust-allocator")]
         if self.zalloc == Allocator::RUST.zalloc {
-            return unsafe { std::alloc::System.alloc_zeroed(layout).cast() };
+            let ptr = unsafe { zalloc_rust_calloc(self.opaque, layout.size() as _, 1) };
```

Comment:
Why is the input alignment now ignored entirely?

Reply:
hmm, I'll add some asserts then. There is little we can do about it, but we know that our allocations never have alignments higher than 64.